clients/horizonclient/error_helpers.go

```diff
@@ -23,6 +23,26 @@ func IsNotFoundError(err error) bool {
 	return hErr.Problem.Type == "https://stellar.org/horizon-errors/not_found"
 }
 
+// IsTimeoutError returns true if the error is a horizonclient.Error with
+// a timeout problem indicating that Horizon timed out.
+func IsTimeoutError(err error) bool {
+	var hErr *Error
+
+	err = errors.Cause(err)
+	switch err := err.(type) {
+	case *Error:
+		hErr = err
+	case Error:
+		hErr = &err
+	}
```

Comment:
We should leverage `GetError()`, below, here:

```suggestion
	hErr := GetError(err)
```

I know the other helper does it this way, but I'd take issue with that, too.

Reply:
https://github.com/stellar/go/pull/3628/commits/efd88f403ac05dd75f25adf41948d7532194218b